bindings/rust/src/lib.rs

```diff
@@ -131,6 +131,84 @@ impl From<f64> for Value {
     }
 }
 
+/// A WebAssembly value i32/i64/f32/f64 with its type specified.
+pub enum TypedValue {
+    U32(u32),
+    U64(u64),
+    F32(f32),
+    F64(f64),
+}
+
+impl TypedValue {
+    fn get_type(&self) -> sys::FizzyValueType {
+        match self {
+            TypedValue::U32(_) => sys::FizzyValueTypeI32,
+            TypedValue::U64(_) => sys::FizzyValueTypeI64,
+            TypedValue::F32(_) => sys::FizzyValueTypeF32,
+            TypedValue::F64(_) => sys::FizzyValueTypeF64,
+        }
+    }
+
+    pub fn as_i32(&self) -> Option<i32> {
+        match self {
+            TypedValue::U32(v) => Some(*v as i32),
```

Comment:
This is still a question, whether we want to do this, or use Rust specific helpers for casting between u32 -> i32.

Reply:
Do you mean the alternative being `x.as_u32() as i32`? I think this is fine then.